### PR TITLE
acrn-config: Reorg config tool folder and category misc content into debug tools and services

### PR DESCRIPTION
**Pull request by NanlinXie**
1.Rename the name of misc sub-folder with "_"
2.Category acrn_crashlog, acrn_log, acrn_trace into debug_tools folder and acrn_bridge, life_mngr, acrn_manager into services.
3.Remove vm_configs folder and move all the XML files and generic code example into config_tools/data

Tracked-On: #5644
Signed-off-by: Xie, nanlin nanlin.xie@intel.com

### CODE REVIEW

**Comment by gvancuts** [comment]
Hi @NanlinXie , is this the complete re-organisation? I'm asking because I thought we wanted to have only subfolders under `misc/config_tools`, called `board`, `hypervisor` and `data`. But I see other folder in there too?

**Comment by NanlinXie** [comment]
@gvancuts , it is the first group of patch for re-organisation. Board and hypervisor need to be created after we revist board.xml and board parser, which most possibly will happen in v2.5.

**Review by gvancuts — APPROVED** [review]
Thanks!

**Comment by gvancuts** [comment]
> @gvancuts , it is the first group of patch for re-organisation. Board and hypervisor need to be created after we revist board.xml and board parser, which most possibly will happen in v2.5.

OK, I did not realize we had to heavily modify this utility.

Is it just the board application that needs to be modified or the others as well?

Just bumped into this line, and wondered if it shouldn't be changed as well? https://github.com/NanlinXie/acrn-hypervisor-1/blob/misc_reorg/misc/config_tools/board_config/board_cfg_gen.py#L20

**Review by dbkinder — APPROVED** [review]
We'll also need to update the documentation that reference commands that are now renamed (with an underscore instead of a dash). I can do that in a separate PR after this is merged.

File: CODEOWNERS [hunk 1]
```diff
@@ -16,13 +16,12 @@ Makefile                                 @terryzouhao
 /hypervisor/                             @anthonyzxu @dongyaozu
 /devicemodel/                            @anthonyzxu @ywan170
 /doc/                                    @dbkinder @deb-intel @NanlinXie
-/misc/tools/acrn-crashlog/               @chengangc @dizhang417 
-/misc/tools/acrnlog/                     @ywan170 @liuyuan-GitHub
-/misc/tools/acrntrace/                   @ywan170 @liuyuan-GitHub
-/misc/acrn-manager/                      @ywan170 @liuyuan-GitHub
-/misc/acrnbridge/                        @ywan170 @liuyuan-GitHub
-/misc/acrn-config/                       @terryzouhao
+/misc/debug_tools/acrn_crashlog/               @chengangc @dizhang417
```
[the remark this answers is not in the document]
Comment (NanlinXie) [reply CODEOWNERS]: Fixed

**Comment by NanlinXie** [comment]
> Is it just the board application that needs to be modified or the others as well?
We will refine the board.xml and board parser to align what information we could get from board, and what board information ACRN need, or hit ACRN assumption to avoid ACRN boot failure.
 
> Just bumped into this line, and wondered if it shouldn't be changed as well? https://github.com/NanlinXie/acrn-hypervisor-1/blob/misc_reorg/misc/config_tools/board_config/board_cfg_gen.py#L20
The script cleanup working will be done in the following PRs.

**Comment by wenlingz** [comment]
@terryzouhao please help approve this PR, thanks

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.

File: CODEOWNERS [hunk 2]
```diff
@@ -16,13 +16,12 @@ Makefile                                 @terryzouhao
...
-/misc/tools/acrnlog/                     @ywan170 @liuyuan-GitHub
-/misc/tools/acrntrace/                   @ywan170 @liuyuan-GitHub
-/misc/acrn-manager/                      @ywan170 @liuyuan-GitHub
-/misc/acrnbridge/                        @ywan170 @liuyuan-GitHub
-/misc/acrn-config/                       @terryzouhao
+/misc/debug_tools/acrn_crashlog/         @chengangc @dizhang417
+/misc/debug_tools/acrn_log/              @ywan170 @shuox
+/misc/debug_tools/acrn_trace/            @ywan170 @shuox
+/misc/services/acrn_manager/             @ywan170 @shuox
+/misc/services/acrn_bridge/              @ywan170 @shuox
```
Comment (gvancuts) [inline CODEOWNERS:23]: Do we have an owner we can already add for these folders?
* `misc/servcies/life_mngr`
* `misc/packaging`

Comment (NanlinXie) [reply CODEOWNERS:23]: I will update that after I confirm the owner.

**Comment by HuiHuangShi** [comment]
No new violations to the coding guideline detected.<br>No New Name Conflict<br>No New Function Declaration/Definition Mismatch

**Review by terryzouhao — APPROVED** [review]

**Pull request merged**